tests/test_utils.py

```diff
@@ -55,3 +56,47 @@ def test_adata_unique_indices():
 
     pd.testing.assert_index_equal(v.obsm["df"].index, v.obs_names)
     pd.testing.assert_index_equal(v.varm["df"].index, v.var_names)
+
+
+def test_adapt_vars_exact_match():
+    # Test that adapt_vars_like works when the source and target have the same var names
+    # and the same number of variables
+    source = ad.AnnData(X=np.ones((1, 3)), var=pd.DataFrame(index=["a", "b", "c"]))
+    target = ad.AnnData(
+        X=np.array([[1, 2, 3]]), var=pd.DataFrame(index=["a", "b", "c"])
+    )
+    output = adapt_vars_like(source, target)
+    np.testing.assert_array_equal(output.X, target.X)
+    assert (output.var.index == target.var.index).all()
+
+
+def test_adapt_vars_different_order():
+    # Test that adapt_vars_like works when the source and target have the same var names
+    # but in a different order
+    source = ad.AnnData(X=np.ones((1, 3)), var=pd.DataFrame(index=["a", "b", "c"]))
+    target = ad.AnnData(
+        X=np.array([[3, 2, 1]]), var=pd.DataFrame(index=["c", "b", "a"])
+    )
+    output = adapt_vars_like(source, target)
+    np.testing.assert_array_equal(output.X, [[1, 2, 3]])
+
+
+def test_adapt_vars_none_X_raises():
+    source = ad.AnnData(X=np.ones((1, 2)), var=pd.DataFrame(index=["g1", "g2"]))
+    target = ad.AnnData(X=None, var=pd.DataFrame(index=["g1", "g2"]))
+    with pytest.raises(ValueError, match="target.X is None"):
+        adapt_vars_like(source, target)
+
+
+def test_adapt_vars_no_shared_genes():
+    source = ad.AnnData(X=np.ones((1, 2)), var=pd.DataFrame(index=["g1", "g2"]))
+    target = ad.AnnData(X=np.array([[7, 8]]), var=pd.DataFrame(index=["g3", "g4"]))
+    output = adapt_vars_like(source, target, fill_value=0.5)
+    np.testing.assert_array_equal(output.X, [[0.5, 0.5]])
+
+
+def test_adapt_vars_missing_genes():
+    source = ad.AnnData(X=np.ones((1, 3)), var=pd.DataFrame(index=["g1", "g2", "g3"]))
+    target = ad.AnnData(X=np.array([[1, 3]]), var=pd.DataFrame(index=["g1", "g3"]))
+    output = adapt_vars_like(source, target, fill_value=-1)
+    np.testing.assert_array_equal(output.X, [[1, -1, 3]])
```

Comment:
These two as well can probably be consolidated as well, maybe with the other two I pointed at, trying everything with both no fill value provided, and a fill value provided.